src/main.cpp

```diff
@@ -11,139 +11,217 @@
-        glfwTerminate();
-        return 1;
-    }
-
-    glfwMakeContextCurrent(window);
-
-    if (glewInit() != GLEW_OK) {
-        std::cerr << "failed to initialize GLEW" << std::endl;
-        return 1;
-    }
-
-    // aci setam ce callbackuri se apeleaza cand se misca mouseu, scrollu sau se apasa pe taste
-    glfwSetCursorPosCallback(window, mouse_callback);
-    glfwSetScrollCallback(window, scroll_callback);
-    glfwSetKeyCallback(window, key_callback);
-    // asta face cursoru sa dispara si sa ramana in fereastra. dai exit sa iesi
-    glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_DISABLED);
-
-    State  state{};
-    Camera camera{{ 0.0f, 0.0f, 0.0f }};
-    // ii dam la window app-stateu, ca sa poata fi accesat in callbackuri
-    glfwSetWindowUserPointer(window, &state);
-
-    // vector de planete. in viitor cel mai probabil aici o sa initializam doar gen soarele sau ce o fi, la care ii adaugam dupa planetele
-    std::vector<Planet> objects;
-    objects.emplace_back(glm::vec3{ -2.0f, 0.0f, -15.0f }, glm::vec3{ 1.0f });
-    objects.emplace_back(glm::vec3{ 2.0f, 0.0f, -15.0f }, glm::vec3{ 1.0f });
-    objects.emplace_back(glm::vec3{ 0.0f, 2.0f, -15.0f }, glm::vec3{ 1.0f, 3.0f, 1.0f });
-
-    // unicu shader. poate o sa mai avem si altele
-    Shader shader{ "shaders/vertex.glsl", "shaders/fragment.glsl" };
-    glUseProgram(shader.id);
-
-    glEnable(GL_DEPTH_TEST);
-    while (!glfwWindowShouldClose(window)) {
-        // asta calculeaza cat timp se scurge intre cadre, ca sa poti sa te folosesti de asta cand misti camera si sa fie mai smooth
-        auto time = (float) glfwGetTime();
-        state.delta.time = time - state.last_time;
-        state.last_time  = time;
-
-        glClearColor(0.5f, 0.5f, 0.5f, 1.0f);
-        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-
-        // updatam camera in functie de ce chestii face useru
-        camera.update(state.held_keys, state.delta);
-
-        glm::mat4 proj{ glm::perspective(glm::radians(camera.fov), (float) WIDTH / HEIGHT, 0.1f, 100.0f) };
-        glm::mat4 view{ camera.view_matrix() };
-
-        // iteram prin obiecte si le calculam la fiecare model matrixu in functie de pozitia si marimea lor, le trimitem la shader, si le dam draw
-        for (const auto& object: objects) {
-            glm::mat4 model{ 1.0f };
-            model = glm::translate(model, object.position);
-            model = glm::scale(model, object.scale);
-            shader.set_mvp(proj * view * model);
-            object.render();
-        }
-
-        glfwSwapBuffers(window);
-        glfwPollEvents();
-    }
-
-    glfwTerminate();
-    return 0;
+void render_satellites_of_satellite(Planet planet, Shader shader, glm::mat4 proj, glm::mat4 view, glm::mat4 model) {
+	for (int i = 0; i < planet.satellites.size(); i++) {
+		Planet satellite = planet.satellites.at(i);
+		model = glm::rotate(model, 8 * (float)glfwGetTime() / satellite.distance, glm::vec3(0.0f, 1.0f, 0.0f));
```

Comment:
cred ca mai bine ii dam la functie un parametru pentru unghiu de rotatie in loc sa luam aici glfwGetTime. ca asa daca o sa avem slider de viteza, o sa putem sa-l integram mai usor. poate n-ar fi rau chiar fiecare planeta sa aiba field de rotation_speed, ca doara n-o sa se invarta toate planetele fix deodata